Created plot_sampling_hist function and tests

YueJiangMDSV opened the pull request:
Hi Team,

In this pr, 
1. created plot_sampling_hist
2. created tests for plot_sampling_hist

Please review it. Thanks!

Alex

YueJiangMDSV committed:
create plot_sampling_hist function

First draft, based on Lise's `draw_samples` function. No tests written yet. Will update with tests. May need to update with any changes with the `draw_samples.py` function

YueJiangMDSV committed:
update the input parameters

YueJiangMDSV committed:
Merge pull request #43 from UBC-MDS/master

catch up with master branch

YueJiangMDSV committed:
update plot_sampling_hist with inputs check

Added test to check the validity of inputs

YueJiangMDSV committed:
update docstring of plot_sampling_hist function

YueJiangMDSV committed:
Create test_plot_sampling_hist.py

YueJiangMDSV committed:
update plot_sampling_hist with tests of the input

Added more tests to test the input of the function

tguo9 commented on samplingsimulatorpy/test_plot_sampling_hist.py:
```diff
@@ -0,0 +1,57 @@
+    --------
+    None
+        The test should pass and no asserts should be displayed. 
+    """
```
```suggestion
```
Probably no needs of docstring here

YueJiangMDSV replied:
Good to know! Thanks!

tguo9 commented on samplingsimulatorpy/plot_sampling_hist.py:
```diff
@@ -0,0 +1,69 @@
+        x=alt.X(f"{summary.columns[3]}:Q", title = "mean of sampling distribution",bin = alt.Bin(extent=[x_min, x_max], step=1/10)),
+        y=alt.Y('count()')).facet(
+        facet='size', columns = 2)
+    return p
```
```suggestion
    return p
    
```
A new line at the end of file

YueJiangMDSV replied:
Just deleted the new line

tguo9 commented on samplingsimulatorpy/test_plot_sampling_hist.py:
```diff
@@ -0,0 +1,57 @@
+    # Check y-axis is mapped to the correct variable and plotting correctly
+    assert test_dict['spec']['encoding']['y']['type'] == 'quantitative', "y-axis should have quantitivate values"
+    assert test_dict['spec']['encoding']['y']['aggregate'] == 'count', "y-axis should represent aggregate count data"
+    
```
```suggestion
```
No need for test file have a new line at the end of file

YueJiangMDSV replied:
Deleted the new line

tguo9 commented on samplingsimulatorpy/plot_sampling_hist.py:
```diff
@@ -0,0 +1,69 @@
+    if not np.issubdtype(samples.to_numpy().dtype, np.number):
+        raise ValueError("'samples' should only contain numeric values")
+        
+    if not x.shape[1]==4:
```
```suggestion
    if not samples.shape[1]==4:
```
x is not defined, I'm guessing it should be samples?

YueJiangMDSV replied:
Good catch! I'll update that

YueJiangMDSV committed:
Update test_plot_sampling_hist.py

Deleted docstring

YueJiangMDSV committed:
Updated the function and tests

Updated plot_sampling_hist function and test_plot_sampling_hist according to Tao's review

YueJiangMDSV committed:
move test function to the test folder

YueJiangMDSV committed:
move the function test_plot_sampling_hist.py to the test folder

tguo9 reviewed (approved):
Looks good! Approve
